modules/apps/portal-instances/portal-instances-test/src/testIntegration/java/com/liferay/portal/instances/internal/configuration/test/PortalInstancesConfigurationFactoryTest.java

```diff
@@ -93,10 +94,36 @@ public void testAddCompanyWithAllAdminProperties() throws Exception {
 	public void testAddCompanyWithDefaultProperties() throws Exception {
 		_testAddCompany(
 			HashMapDictionaryBuilder.<String, Object>put(
+				"adminEmailAddress", "testAdminEmailAddress@" + _domain
+			).put(
+				"mx", _domain
+			).put(
+				"virtualHostname", _domain
+			).build());
+	}
+
+	@Test
+	public void testAddCompanyWithInvalidEmail() throws Exception {
+		boolean companyCreated = true;
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"adminEmailAddress", "testAdminEmailAddressAT" + _domain
+			).put(
 				"mx", _domain
 			).put(
 				"virtualHostname", _domain
 			).build());
+
+		try {
+			_company = _companyLocalService.getCompanyByWebId(_webId);
+		}
+		catch (NoSuchCompanyException noSuchCompanyException) {
+			companyCreated = false;
+		}
+
+		Assert.assertFalse(companyCreated);
```

Comment:
@devopsmystic 

Firstly, for the current assertion, a better way is to use an `Assert.fail()` in the try block:
```
		try {
			_company = _companyLocalService.getCompanyByWebId(_webId);
			
			Assert.fail("reason");
		}
		catch (NoSuchCompanyException noSuchCompanyException) {
			// swallow it, or assert the message of the NSCE if SF complains about the exception not being handled
		}
```
With this, it fails immediately if the `getCompanyByWebId()` method doesn't throw any exception; no need to use a boolean variable.

Additionally, to make the assertion even more complete, use `LoggerTestUtil` to wrap whole logic and assert that the info log is correctly printed.

Reply:
By the way, please add ticket number to this commit.